src/Command/PelagosMigrateDatasetFilesCommand.php

```diff
@@ -0,0 +1,209 @@
+                $this->setFile($datasetSubmission, $dataStore, $queueFiler);
+            }
+
+            $this->entityManager->persist($dataset);
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+        $io->newLine(3);
+
+        $io->section('Flushing');
+        $this->entityManager->flush();
+
+        $io->section('Queuing Filer');
+        $this->submissionsListForFiler = array_unique($this->submissionsListForFiler);
+        $io->note(sprintf('Files For Queue %d.', count(($this->submissionsListForFiler))));
+
+        foreach ($this->submissionsListForFiler as $datasetSubmissionId) {
+            $datasetSubmissionFilerMessage = new DatasetSubmissionFiler($datasetSubmissionId);
+            $this->messageBus->dispatch($datasetSubmissionFilerMessage);
+        }
+
+        $io->success('Done!');
+
+        return 0;
+    }
+
+    /**
+     * Set file data for the dataset submission.
+     *
+     * @param DatasetSubmission  $datasetSubmission The Dataset Submission to process.
+     * @param String             $dataStore         The path to the datatore.
+     * @param Boolean            $queueFiler        Should this also be queued to the filer.
+     */
+    protected function setFile(
+        DatasetSubmission $datasetSubmission = null,
+        String $dataStore,
+        bool $queueFiler
+    ) {
+        if ($datasetSubmission instanceof DatasetSubmission) {
+            $udi = $datasetSubmission->getDataset()->getUdi();
+
+            $fileName = "$dataStore/$udi/$udi.dat";
+
+            if (!file_exists($fileName)) {
+                return;
+            }
+
+            $fileData = $this->getFileData($datasetSubmission);
+
+            $file = new File();
+            $file->setFilePathName($fileData['datasetFileName']);
+            $file->setFileSize($fileData['datasetFileSize']);
+            $file->setFileSha256Hash($fileData['datasetFileSha256Hash']);
+
+            $file->setUploadedAt($datasetSubmission->getModificationTimeStamp());
+            $file->setUploadedBy($datasetSubmission->getModifier());
+
+            $file->setDescription('Original Dataset File');
+
+            $file->setPhysicalFilePath($fileName);
+            $file->setStatus($queueFiler ? File::FILE_NEW : File::FILE_DONE);
+
+            $fileset = $datasetSubmission->getFileset();
+
+            $newFileset = new Fileset();
+            $newFileset->addFile($file);
+            $datasetSubmission->setFileset($newFileset);
+
+            if ($queueFiler === true) {
+                $this->submissionsListForFiler[] = $datasetSubmission->getId();
+            }
+        }
+    }
+
+    /**
+     * Get the file data from the submission, because the settings have been forwarded.
+     *
+     * @param DatasetSubmission $datasetSubmission The Dataset Submission to process.
+     */
+    protected function getFileData($datasetSubmission)
+    {
+        $datasetSubmissionid = $datasetSubmission->getid();
```

Comment:
isn't it getId()?

Reply:
Yes, but how does it work then? I think it's some kind of "magic method" because id is always part of the entity.
Anyway, it's changed.